docs/config_reference.rst

```diff
@@ -783,6 +783,16 @@ They are associated with `system partitions <#system-partition-configuration>`__
    .. versionadded:: 3.9.1
 
 
+.. py:attribute:: environments.prepare_cmds
+
+   :required: No
+   :default: ``[]``
+
+   List of shell commands to be emitted after the :attr:`~config.systems.partitions.prepare_cmds` but before any module loading commands(if applicable) and environment variables are set.
```

Comment:
I would not correlate this with the partition semantically. Whatever is defined in the environment should be scoped to it. So better rephrase it as:

```suggestion
   List of shell commands to be emitted before any commands that load the environment.
```